restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java

```diff
@@ -1467,25 +1577,37 @@ void onNotificationIntent(Intent intent)
             PackageManager packageManager = context.getPackageManager();
             actionIntent = packageManager.getLaunchIntentForPackage(context.getPackageName());
          }
+         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            stopForeground(true);
+         }
 
       }
       else if (intentAction.equals(ACTION_NOTIFICATION_CALL_ACCEPT_VIDEO)) {
          callIntent.setAction(ACTION_INCOMING_CALL_ANSWER_VIDEO);
          // don't forget to copy the extras
          callIntent.putExtras(intent);
          actionIntent = callIntent;
+         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            stopForeground(true);
+         }
       }
       else if (intentAction.equals(ACTION_NOTIFICATION_CALL_ACCEPT_AUDIO)) {
          callIntent.setAction(ACTION_INCOMING_CALL_ANSWER_AUDIO);
          // don't forget to copy the extras
          callIntent.putExtras(intent);
+         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            stopForeground(true);
+         }
          actionIntent = callIntent;
       }
       else if (intentAction.equals(ACTION_NOTIFICATION_CALL_DECLINE) || intentAction.equals(ACTION_NOTIFICATION_CALL_DELETE)) {
          RCConnection pendingConnection = getPendingConnection();
          if (pendingConnection != null) {
             pendingConnection.reject();
          }
+
+         release();
```

Comment:
Is there another way the user has to not accept the call other than decline? Can they somehow ignore? Trying to make sure that we cover all cases where release() should be called

Reply:
I think I bumped on such a case, when the use swipes right or left on the call. What would happen then?

Reply:
We should check onTaskRemoved().